src/components/WwwFrame/LendMenu/TheLendMenu.vue

```diff
@@ -107,48 +130,30 @@ export default {
 		onLoad() {
 			this.apollo.watchQuery({ query: publicLendMenuQuery }).subscribe({
 				next: ({ data }) => {
-					this.userId = _get(data, 'my.userAccount.id');
 					this.categories = _get(data, 'lend.loanChannels.values');
-					this.countryFacets = _get(data, 'lend.countryFacets');
+					this.isApolloLoading = false;
 				}
 			});
 		},
-		startLoading() {
-			this.loadingSemaphore += 1;
-		},
-		stopLoading() {
-			if (this.loadingSemaphore > 0) {
-				this.loadingSemaphore -= 1;
-			} else {
-				this.loadingSemaphore = 0;
-			}
-		},
 	},
-	watch: {
-		hasUserId(hasUserId) {
```

Comment:
I assume this was maybe left over from before auth0? The userId seems to be reliably available at prefetch, so not sure why we need the watch and the loadingSemaphore...

Reply:
The watch is necessary to handle the user becoming logged in while still in the page, so that when they open the menu again they get the personalization

Reply:
Can you give me an example of where this isn't working? I tested this code using this flow: 

1. Have something in basket
2. be logged out
3. go to `/checkout`
4. open lend menu
5. click 'Sign In' to trigger log in pop up
6. sign in
7. open lend menu again

Reply:
Yeah that makes sense since using the apollo options sets up the watch for you.